src/command/marian_adaptive_server.cpp

```diff
@@ -0,0 +1,60 @@
+#include "marian.h"
+
+#include "3rd_party/simple-websocket-server/server_ws.hpp"
+#include "common/file_stream.h"
+#include "common/timer.h"
+#include "common/utils.h"
+#include "training/training.h"
+#include "translator/self_adaptive.h"
+
+using namespace marian;
+
+typedef SimpleWeb::SocketServer<SimpleWeb::WS> WSServer;
+
+int main(int argc, char **argv) {
+  auto options = parseOptions(argc, argv, cli::mode::selfadaptiveServer);
+  auto task = New<TrainSelfAdaptive>(options);
```

Comment:
This entire file looks like an old version of `marian_server.cpp` with only these two lines different. I would suggest checking if the two files can be combined (might not be straightforward) or please update the rest of the file so that it looks the same as `marian_server.cpp` if possible.

Reply:
Do you think it's better to handle adaptive and non-adaptive server stuff in marian_server.cpp or factor out the common parts into a separate file that's called from both marian_server.cpp and marian_adaptrive_server.cpp?

I'd prefer the 2nd option, i guess. Combining the two would require adding a CLI option to distinguish among regular and self-adaptive translation.

Reply:
I've implemented the 2nd option.